src/mgr/DaemonServer.cc

```diff
@@ -1979,7 +1979,9 @@ bool DaemonServer::handle_command(MCommand *m)
     }
 
     std::stringstream ds;
-    int r = py_modules.handle_command(handler_name, cmdctx->cmdmap, &ds, &ss);
+    bufferlist inbl = cmdctx->m->get_data();
+    string instr(inbl.c_str(), inbl.length());
```

Comment:
This becomes a `char*` on the way to python, so I think we need to sanitize it for null bytes

Reply:
Shower thinking leads me to believe it should go to Python as a counted non-NUL-terminated string anyway.  Will redo.

Reply:
It isn't unsafe per se, right, since any null bytes just shorten the string?  Is there a way to pass the full buffer to python?

Reply:
I think it would be better to pass around the bufferlist until the submission to Python anyway, and then pass a counted string (since we have the length without further scanning, and since I think Python can handle strings with embedded NULs, although of course if it resubmits it to Ceph there could be trouble)